src/test/java/echoServer/MockClientWriterFactory.java

```diff
@@ -0,0 +1,25 @@
+package echoServer;
+
+import echoServer.outputManagement.ClientWriteable;
+import echoServer.outputManagement.ClientWriterInterface;
+
+import java.io.IOException;
+import java.net.Socket;
+
+public class MockClientWriterFactory implements ClientWriteable {
+    
+    public static Socket socket;
+    private final ClientWriterInterface printer;
+    boolean printWriterWasCalled = false;
+
+    public MockClientWriterFactory(ClientWriterInterface printer) {
+        this.printer = printer;
+    }
+
+    @Override
+    public ClientWriterInterface makePrinter(Socket clientSocket) throws IOException {
+        boolean printWriterWasCalled = true;
```

Comment:
We actually don't have any function called `printWriter`. I don't think any of the attributes on this mock are necessary since you don't assert against them.

Reply:
I did need the printer in order to make one of my tests run, but I removed the other two.